WordPressKitTests/AllDomainsResultDomainTests.swift

```diff
@@ -0,0 +1,160 @@
+import Foundation
+import XCTest
+
+@testable import WordPressKit
+
+final class AllDomainsResultDomainTests: XCTestCase {
+
+    // MARK: - Properties
+
+    private let dateFormatter = ISO8601DateFormatter()
+
+    // MARK: - Tests
+
+    func testDecoding1() throws {
```

Comment:
[Suggestion] I suggest using more descriptive test names. I see each test is testing a different form of input. Maybe we can use name that reflect the differences.

Reply:
@hassaanelgarem I agree. I spent a lot of time trying to come up with descriptive test names, but I wasn't happy with the result. Here is what I tried:

- Before `testDecoding1`, it was `testDecodingWithDefaultInput`.
- Before `testDecoding2`, it was `testDecodingWithEmptyExpiryDateAndRegistrationDate`
- Before `testDecoding3`, it was `testDecodingWithMappingTypeAndHasRegistrationIsTrue`

The difference between the test cases is the input, the idea was to test the decoding logic with different input. And it's difficult to come with test names describing the difference.

Reply:
I think having long test names is more than fine. I like the names you originally thought of 👍

Reply:
Resolved in 4328bad561ac715bfb377695aa2397b016a3bfef

Reply:
Something I've seen in the Apple docs is also to supplement test names that might be hard to read with a documentation comment (`///`, not `//`):

```swift
/// When expiry date and registration date are given,
/// Then the output is ...
func testDecodingWithEmptyExpiryDateAndRegistrationDate() {
```